Preprocessing

devyansh opened the pull request.

devyansh committed:
working on windows now

devyansh committed:
Merge branch 'master' of https://github.com/saketh-n/seizure-classification

devyansh committed:
plots

minneker commented on Pre-processing/PlotAllChannels.py:
```diff
@@ -0,0 +1,21 @@
+
+
+#Load Data
+load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/00003010_s003_t000.edf'
```
local path needs to change, won't work on another computer.

minneker commented on Pre-processing/PlotOneChannel.py:
```diff
@@ -0,0 +1,44 @@
+
+
+#Load Data
+load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/00003010_s003_t000.edf'
```
same here, need to point to file not an absolute path.

minneker commented on Pre-processing/PlotOneChannel.py:
```diff
@@ -0,0 +1,44 @@
+raw = read_raw_edf(load_path, preload=True)
+
+#Inspect the .edf file
+print(raw.info['ch_names'])
```
do we need to inspect the file when running? Or are these just for debugging? If the latter, I think we can remove all inspection print statements.

devyansh committed:
Update PlotOneChannel.py

minneker commented on package.json:
```diff
@@ -6,17 +6,18 @@
     "build:css": "postcss src/styles/tailwind.css -o src/styles/app.css",
     "watch:css": "postcss src/styles/tailwind.css -o src/styles/app.css --watch",
-    "react-scripts:start": "sleep 5 && react-scripts start",
+    "react-scripts:start": "timeout 5 && react-scripts start",
```
windows specific changes should be handled dynamically and not hardcoded. We should think about this carefully before merging to master.

saketh-n replied:
agreed.

minneker commented on yarn.lock:
```diff
@@ -146,7 +146,7 @@
   "version" "7.13.13"
   dependencies:
-    "@babel/compat-data" "^7.13.12"
+    "@babel/compat-data" "^7.13.15"
```
same with these changes. We should probably have an OS-specific switch that determines the correct versions here.

devyansh committed:
generating plots for each channel individually

2 plots for each Channel: one for RAW Data and other is Spectral Power Density

PlotAllChannels plots and saves the images as well

saketh-n commented on Pre-processing/PlotAllChannels.py:
```diff
@@ -1,21 +1,38 @@
 
 #Load Data
-load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/00003010_s003_t000.edf'
+filename = '00003010_s003_t000'
```
I guess it's fine for now, but when I integrate with front end I want to pull this filename from the api call instead of hardcoded

saketh-n commented on Pre-processing/PlotAllChannels.py:
```diff
@@ -1,21 +1,38 @@
 #Load Data
-load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/00003010_s003_t000.edf'
+filename = '00003010_s003_t000'
+fileformat = '.edf'
```
Idk if this would be necessary if filename passed through api call

saketh-n commented on Pre-processing/PlotAllChannels.py:
```diff
@@ -1,21 +1,38 @@
-load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/00003010_s003_t000.edf'
+filename = '00003010_s003_t000'
+fileformat = '.edf'
+load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/' + filename + fileformat
```
In general relative paths are preferred, but passing in filename should make this a non-issue. When web-hosting this will be a more interesting problem. We should try to have pre-processing code run on the front end

saketh-n commented on Pre-processing/PlotAllChannels.py:
```diff
@@ -1,21 +1,38 @@
+filename = '00003010_s003_t000'
+fileformat = '.edf'
+load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/' + filename + fileformat
 print(load_path)
```
We should remove debugging code when we integrate w master

saketh-n commented on Pre-processing/PlotAllChannels.py:
```diff
@@ -1,21 +1,38 @@
 
+#Make a directory
+directory = filename + '_allChannels'
+parent_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/GeneratedPlots/'
```
Nice! once again relative paths, maybe something like os.pwd()

saketh-n commented on Pre-processing/PlotAllChannels.py:
```diff
@@ -1,21 +1,38 @@
 #Raw data
-raw.plot(duration=100, block=True)
-plt.show()
+#todo: instead of scrolly image, get a static image
```
If we get this running on the front end, we might not even need to save anything! Tricky problem though since browser only runs js/css/html

saketh-n commented on Pre-processing/PlotOneChannel.py:
```diff
@@ -1,35 +1,51 @@
 
 #Load Data
-load_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/TUH/00003010_s003_t000.edf'
+filename = '00003010_s003_t000'
```
Same note as above, relative paths + filename either passed in, or raw data is

saketh-n commented on Pre-processing/PlotOneChannel.py:
```diff
@@ -1,35 +1,51 @@
+
+#Make a directory
+directory = filename
+parent_path = 'C:/Users/User/Desktop/CSE 481C - Neruo Capstone/Project Repo/Data/GeneratedPlots/'
```
relative path

devyansh committed:
Create PlotAllChannelsScrolly.py

Plots RAW data of all channels in an interactive GUI

devyansh committed:
creating Epochs for TFRs

minneker commented on Pre-processing/epochs.py:
```diff
@@ -0,0 +1,69 @@
+import os
```
This code is largely taken from: https://mne.tools/mne-realtime/_modules/mne/epochs.html.

Please make sure to cite any code you copy-paste.

minneker commented on .idea/misc.xml:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
All of these .idea files are not necessary. They help render the project in your local IDE. Please remove any .idea/ files. We should add this to .gitignore to avoid in future pushes.

minneker commented on Pre-processing/epochs.py:
```diff
@@ -0,0 +1,69 @@
+import os
+import mne
+import numpy as np
```
Also this code seems to already exists in the MNE package. Why are we writing it locally instead of calling it from the package?

devyansh committed:
All desired plots generated

RAW of all channels
TFR of all channels
SPD of all channels

devyansh committed:
shifted functions to different file

also changed the relative path of parent_path

devyansh committed:
parent_path update

devyansh committed:
Update main.py
